treeherder/perf/perf_sheriff_bot.py

```diff
@@ -15,3 +21,17 @@ def report(self, since: datetime,
                frameworks: List[str],
                repositories: List[str]) -> List[BackfillReport]:
         return self.report_maintainer.provide_updated_reports(since, frameworks, repositories)
+
+    def _queue_is_too_loaded(self, provisioner_id: str, worker_type: str, acceptable_limit=1000) -> bool:
```

Comment:
I think something like `check_queue_load` or `is_queue_overloaded` would be clearer.

Does the `acceptable_limit=1000` mean 1000 messages in the queue? I'm curious how you determined 1000 to be the max limit.

Reply:
It's arbitrary. 
I will give today a more precise number, based on various random Treeherder jobs.
I'll look at 100 random jobs and see what's the max queue load.
Anyway, this number can be adjusted later.
For the moment I'm interested in not blocking the Bot.
After Bot integration, we'll supervise its activity. 
If things go wild, we'll stop the Bot and make required adaptations.

Reply:
As of Monday, 2:00 PM (Romania time), the results are:
```
provisioner_id: gecko-3, worker_type: gecko-t-linux-talos, pending_tasks_count: 0
provisioner_id: gecko-3, worker_type: gecko-t-signing, pending_tasks_count: 0
provisioner_id: gecko-3, worker_type: t-linux-large, pending_tasks_count: 0
provisioner_id: gecko-3, worker_type: t-linux-xlarge, pending_tasks_count: 0
provisioner_id: gecko-3, worker_type: b-linux, pending_tasks_count: 0
provisioner_id: gecko-3, worker_type: b-linux-aws, pending_tasks_count: 0
provisioner_id: gecko-t, worker_type: gecko-t-linux-talos, pending_tasks_count: 0
provisioner_id: gecko-t, worker_type: gecko-t-signing, pending_tasks_count: 0
provisioner_id: gecko-t, worker_type: t-linux-large, pending_tasks_count: 43
provisioner_id: gecko-t, worker_type: t-linux-xlarge, pending_tasks_count: 0
provisioner_id: gecko-t, worker_type: b-linux, pending_tasks_count: 0
provisioner_id: gecko-t, worker_type: b-linux-aws, pending_tasks_count: 0
provisioner_id: scriptworker-k8s, worker_type: gecko-t-linux-talos, pending_tasks_count: 0
provisioner_id: scriptworker-k8s, worker_type: gecko-t-signing, pending_tasks_count: 0
provisioner_id: scriptworker-k8s, worker_type: t-linux-large, pending_tasks_count: 0
provisioner_id: scriptworker-k8s, worker_type: t-linux-xlarge, pending_tasks_count: 0
provisioner_id: scriptworker-k8s, worker_type: b-linux, pending_tasks_count: 0
provisioner_id: scriptworker-k8s, worker_type: b-linux-aws, pending_tasks_count: 0
provisioner_id: releng-hardware, worker_type: gecko-t-linux-talos, pending_tasks_count: 0
provisioner_id: releng-hardware, worker_type: gecko-t-signing, pending_tasks_count: 0
provisioner_id: releng-hardware, worker_type: t-linux-large, pending_tasks_count: 0
provisioner_id: releng-hardware, worker_type: t-linux-xlarge, pending_tasks_count: 0
provisioner_id: releng-hardware, worker_type: b-linux, pending_tasks_count: 0
provisioner_id: releng-hardware, worker_type: b-linux-aws, pending_tasks_count: 0
```
I'll wait until tomorrow morning and see the new values (after folks come to work and do git pushes), in order to select a more precise value.

Reply:
Just checked. The results (Tuesday morning, Romania time):
```
provisioner_id: scriptworker-k8s, worker_type: gecko-t-signing, pending_tasks_count: 1
```
All other queues have 0 `pending_tasks_count`.
I'll check in the night again.

Reply:
Tuesday night update (investigated randomly in time):
```
provisioner_id: gecko-3, worker_type: b-linux, pending_tasks_count: 74
provisioner_id: gecko-t, worker_type: t-linux-large, pending_tasks_count: 32
provisioner_id: gecko-3, worker_type: b-linux, pending_tasks_count: 92
provisioner_id: gecko-t, worker_type: t-linux-large, pending_tasks_count: 58
provisioner_id: gecko-3, worker_type: b-linux, pending_tasks_count: 130
provisioner_id: gecko-t, worker_type: t-linux-large, pending_tasks_count: 52
provisioner_id: gecko-t, worker_type: t-linux-large, pending_tasks_count: 137
provisioner_id: gecko-t, worker_type: t-linux-xlarge, pending_tasks_count: 24
provisioner_id: gecko-t, worker_type: t-linux-large, pending_tasks_count: 278
provisioner_id: gecko-t, worker_type: t-linux-xlarge, pending_tasks_count: 116
```

Reply:
Interesting findings!
@octavian-negru how about adding a bit of extra logic? Like having some sort of mapping between `provisioner`+`worker_type` -> 2 * `pendings_task_count`. In case a key is missing, we default to 70.